core/controllers/profile.py

```diff
@@ -361,6 +364,7 @@ def post(self):
 class SiteLanguageHandler(base.BaseHandler):
     """Changes the preferred system language in the user's preferences."""
 
+    @acl_decorators.open_access
```

Comment:
That seems weird to me. Why don't we just not send requests to this handler from non-logged-in users?